doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -92,6 +92,155 @@ writeShellScript "evaluate-my-file.sh" ''
+
+This function is most often used with `mkDerivation.desktopItems` to add desktop items to a package.
+
+`makeDesktopItem` takes an attribute set according to the [XDG specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys), with the following possible values:
+
+Note that the 'Version' and 'Hidden' fields are not supported.
+
+`name` (String)
+
+: The name of the desktop file in the Nix store.
+
+`type` (String, _optional_)
+
+: According to the specification.
+
+  Default: `"Application"`
+
+`desktopName` (String)
+
+: According to the specification.
+
+`genericName` (String, _optional_)
+
+: According to the specification.
+
+`noDisplay` (Bool, _optional_)
+
+: According to the specification.
+
+`comment` (String, _optional_)
+
+: According to the specification.
+
+`icon` (String, _optional_)
+
+: According to the specification.
+
+`onlyShowIn` (List of String, _optional_)
+
+: According to the specification.
+
+`notShowIn` (List of String, _optional_)
+
+: According to the specification.
+
+`dbusActivatable` (Bool, _optional_)
+
+: According to the specification.
+
+`tryExec` (String, _optional_)
+
+: According to the specification.
+
+`exec` (String, _optional_)
+
+: According to the specification.
+
+`path` (String, _optional_)
+
+: According to the specification.
+
+`terminal` (Bool, _optional_)
+
+: According to the specification.
+
+`actions` (List of Attribute set, _optional_)
+
+: According to the specification.
+
+  A list of attribute sets {name, exec?, icon?}
+
+`mimeTypes` (List of String, _optional_)
+
+: According to the specification.
+
+`categories` (List of String, _optional_)
+
+: According to the specification.
+
+`implements` (List of String, _optional_)
+
+: According to the specification.
+
+`keywords` (List of String, _optional_)
+
+: According to the specification.
+
+`startupNotify` (Bool, _optional_)
+
+: According to the specification.
+
+`startupWMClass` (String, _optional_)
+
+: According to the specification.
+
+`url` (String, _optional_)
+
+: According to the specification.
+
+`prefersNonDefaultGPU` (Bool, _optional_)
+
+: According to the specification.
+
+  Note that this is not supported until version 1.5, which the `desktop-file-utils` in Nixpkgs does not support.
+
+`extraConfig` (Attribute set, _optional_)
+
+: Additional key/value pairs to be added verbatim to the desktop file. Attributes need to be prefixed with 'X-'.
+
+::: {.example #ex-makeDesktopItem}
+# Usage of `makeDesktopItem`
+
+Write a desktop file `/nix/store/<store path>/my-program.desktop` to the Nix store.
+
+```nix
+makeDesktopItem {
+  name = "my-program";
+  desktopName = "My Program";
+  genericName = "Video Player";
+  noDisplay = false;
+  comment = "Cool video player";
+  icon = "/path/to/icon";
+  onlyShowIn = [ "KDE" ];
+  dbusActivatable = true;
+  tryExec = "myprogram";
+  exec = "myprogram --someflag";
```

Comment:
I'm not too familiar with these desktop files, but seems a pattern to not pass the complete path to the executable in them. Do you think this is important to document, or would things still work if someone decided to do something like:
```
exec = "${pkgs.hello}"
```

Reply:
Yeah, that way is better for purity, since using just the executable name might end up using some other executable.

Reply:
If you think it's important to document this, please do! When searching for usage of `makeDesktopItem` across Nixpkgs, I found many cases that don't actually use the full path, so I was unsure.

Reply:
Changed these to full paths, also added a note about this.

Maybe a bigger PR that changes these to full paths is also in order.